com.playeveryware.eos/Runtime/Core/Common/SetOfNamed.cs

```diff
@@ -55,6 +64,28 @@ public SetOfNamed(string defaultNamePattern = "NamedItem")
             _defaultNamePattern = defaultNamePattern;
         }
 
+        public void SetRemovePredicate(Func<T, bool> removePredicate)
+        {
+            _removePredicate = removePredicate;
+        }
+
+        /// <summary>
+        /// Returns the name of a new item to be added to the collection.
+        /// </summary>
+        /// <returns>The name to give to the added set.</returns>
+        private string GetNewItemName()
+        {
+            string newItemName = _defaultNamePattern;
+            int increment = 0;
```

Comment:
I feel strange about this code; doing string interpolation and comparison continually sounds more expensive than needed. Could we just keep `increment` around and keep adding to that?

Reply:
Probably good to relocate this into an "Editor" directory to make it clear that something so non-performant is only going to execute in the editor.

Reply:
Instead of taking that approach, I changed the implementation to use a binary search algorithm to determine the increment to use for the new name. The maximum increment is equal to the number of items in the collection already, and the minimum increment is one. The new implementation performs a binary search between the two values and continues until the smallest increment is determined.

This approach was favored over moving the collection to an editor-only context because the code that performs migration makes use of SetOfNamed, and the migration code is not guaranteed to only run in the editor (by design).